extensions/mqtt/processors/ConsumeMQTT.cpp

```diff
@@ -34,111 +34,333 @@ void ConsumeMQTT::initialize() {
+    session->putAttribute(flow_file, key, value);
+  }
+}
+
+void ConsumeMQTT::fillAttributeFromContentType(const SmartMessage& message, const std::shared_ptr<core::FlowFile>& flow_file, const std::shared_ptr<core::ProcessSession>& session) const {
+  if (mqtt_version_.value() != MqttVersions::V_5_0 || attribute_from_content_type_.empty()) {
+    return;
+  }
+
+  MQTTProperty* property = MQTTProperties_getProperty(&message.contents->properties, MQTTPROPERTY_CODE_CONTENT_TYPE);
+  if (property == nullptr) {
+    return;
+  }
+
+  std::string content_type(property->value.data.data, property->value.data.len);
+  session->putAttribute(flow_file, attribute_from_content_type_, content_type);
 }
 
-bool ConsumeMQTT::startupClient() {
+void ConsumeMQTT::startupClient() {
   MQTTAsync_responseOptions response_options = MQTTAsync_responseOptions_initializer;
   response_options.context = this;
-  response_options.onSuccess = subscriptionSuccess;
-  response_options.onFailure = subscriptionFailure;
-  const int ret = MQTTAsync_subscribe(client_, topic_.c_str(), gsl::narrow<int>(qos_), &response_options);
+
+  if (mqtt_version_.value() == MqttVersions::V_5_0) {
+    response_options.onSuccess5 = subscriptionSuccess5;
+    response_options.onFailure5 = subscriptionFailure5;
+  } else {
+    response_options.onSuccess = subscriptionSuccess;
+    response_options.onFailure = subscriptionFailure;
+  }
+
+  const int ret = MQTTAsync_subscribe(client_, topic_.c_str(), gsl::narrow<int>(qos_.value()), &response_options);
   if (ret != MQTTASYNC_SUCCESS) {
     logger_->log_error("Failed to subscribe to MQTT topic %s (%d)", topic_, ret);
-    return false;
+    return;
   }
   logger_->log_debug("Successfully subscribed to MQTT topic: %s", topic_);
-  return true;
 }
 
-void ConsumeMQTT::onMessageReceived(char* topic_name, int /*topic_len*/, MQTTAsync_message* message) {
-  MQTTAsync_free(topic_name);
+void ConsumeMQTT::onMessageReceived(SmartMessage smart_message) {
+  if (mqtt_version_ == MqttVersions::V_5_0) {
+    resolveTopicFromAlias(smart_message);
+  }
+
+  if (smart_message.topic.empty()) {
+    logger_->log_error("Received message without topic");
+    return;
+  }
+
+  enqueueReceivedMQTTMsg(std::move(smart_message));
+}
+
+void ConsumeMQTT::resolveTopicFromAlias(SmartMessage& smart_message) {
+  auto raw_alias = MQTTProperties_getNumericValue(&smart_message.contents->properties, MQTTPROPERTY_CODE_TOPIC_ALIAS);
+
+  std::optional<uint16_t> alias;
+  if (raw_alias != PAHO_MQTT_C_FAILURE_CODE) {
+    alias = gsl::narrow<uint16_t>(raw_alias);
+  }
+
+  auto& topic = smart_message.topic;
 
-  const auto* msgPayload = reinterpret_cast<const char*>(message->payload);
-  const size_t msgLen = message->payloadlen;
-  const std::string messageText(msgPayload, msgLen);
-  logger_->log_debug("Received message \"%s\" to MQTT topic %s on broker %s", messageText, topic_, uri_);
+  if (alias.has_value()) {
+    if (*alias > topic_alias_maximum_) {
```

Comment:
I see, thanks!  I thought "Maximum number of topic aliases to use, set to 0 to turn feature off." in the documentation of `Topic Alias Maximum` meant that if set to 0, then there is no maximum.  Since this behavior is surprising, I think we should change the documentation string to something clearer, eg. "Maximum number of topic aliases to use.  If set to 0, then topic aliases cannot be used."